src/errors/checkXcmTxInputs.ts

```diff
@@ -380,22 +552,145 @@ const checkParaToSystemAssetId = (
+	api: ApiPromise,
 	assetId: string,
 	specName: string,
 	relayChainInfo: ChainInfo,
 	registry: Registry,
-	xcmDirection: Direction
+	xcmDirection: Direction,
+	isForeignAssetsTransfer: boolean
 ) => {
-	checkIsValidSystemChainAssetId(
+	await checkIsValidSystemChainAssetId(
+		api,
 		assetId,
 		specName,
 		relayChainInfo,
 		registry,
-		xcmDirection
+		xcmDirection,
+		isForeignAssetsTransfer
 	);
 };
 
+/**
+ * Checks to ensure that assetId's have a length no greater than MAX_ASSETS_FOR_TRANSFER, throws an error if greater than MAX_ASSETS_FOR_TRANSFER
+ *
+ * @param assetIds
+ */
+export const checkAssetIdsLengthIsValid = (assetIds: string[]) => {
+	if (assetIds.length > MAX_ASSETS_FOR_TRANSFER) {
+		throw new BaseError(
+			`Maximum number of assets allowed for transfer is 2. Found ${assetIds.length} assetIds`
+		);
+	}
+};
+
+/**
+ * Checks to ensure that assetId's have no duplicates, throws an error if duplicate is found
+ *
+ * @param assetIds
+ */
+export const checkAssetIdsHaveNoDuplicates = (assetIds: string[]) => {
+	const duplicateAssetIds: string[] = [];
+
+	if (assetIds.length > 1) {
+		for (let i = 0; i < assetIds.length; i++) {
+			const asset1 = assetIds[i];
+
+			for (let j = 0; j < assetIds.length; j++) {
+				if (i === j) {
+					continue;
+				}
+
+				const asset2 = assetIds[j];
+				if (
+					asset1.trim().toLowerCase().replace(/ /g, '') ===
+					asset2.trim().toLowerCase().replace(/ /g, '')
+				) {
+					duplicateAssetIds.push(asset2);
+				}
+			}
+		}
+
+		if (duplicateAssetIds.length > 0) {
+			if (assetIds[0] === '') {
+				throw new BaseError(
+					`AssetIds must be unique. Found duplicate native relay assets as empty strings`
+				);
+			}
+
+			throw new BaseError(
+				`AssetIds must be unique. Found duplicate assetId ${duplicateAssetIds[0]}`
+			);
+		}
+	}
+};
+
+/**
+ * Checks to ensure that assetId's are symbols, integers and empty string or multilocations exclusively
+ *
+ * @param assetIds
+ */
+export const checkAssetIdsAreOfSameAssetIdType = (assetIds: string[]) => {
+	if (assetIds.length > 1) {
+		let relayDefaultValueFound = false;
+		let symbolAssetIdFound = '';
+		let integerAssetIdFound = '';
+		let multiLocationAssetIdFound = '';
+
+		for (const assetId of assetIds) {
+			if (assetId === '') {
+				relayDefaultValueFound = true;
+				continue;
+			}
+
+			const parsedAssetIdAsNumber = Number.parseInt(assetId);
+			const isNotANumber = Number.isNaN(parsedAssetIdAsNumber);
+
+			if (!isNotANumber) {
+				integerAssetIdFound = assetId;
+			} else if (assetId.toLowerCase().includes('parents')) {
+				multiLocationAssetIdFound = assetId;
+			} else {
+				symbolAssetIdFound = assetId;
+			}
+		}
+
+		if (relayDefaultValueFound && multiLocationAssetIdFound) {
+			throw new BaseError(
+				`Found both default relay native asset and multilocation assetId: ${multiLocationAssetIdFound}. Asset Ids must be either be a combination of empty string, symbol and integer or multilocations exclusively.`
```

Comment:
Suggestion for error message: "Found both default relay native asset and foreign asset with assetId: ${multiLocationAssetIdFound}. Relay native asset and foreign assets can't be transferred within the same call."

Reply:
same comment for the spec file